atable/src/components/ATableColumnFilter.vue

```diff
@@ -0,0 +1,212 @@
+<template>
+	<div class="column-filter">
+		<input
+			v-if="getFilterType(column) === 'text'"
+			v-model="filterValue"
+			type="text"
+			class="filter-input"
+			@input="updateFilter('text', filterValue)" />
+
+		<input
+			v-else-if="getFilterType(column) === 'number'"
+			v-model="filterValue"
+			type="number"
+			class="filter-input"
+			@input="updateFilter('number', filterValue)" />
+
+		<label v-else-if="getFilterType(column) === 'checkbox'" class="checkbox-filter">
+			<input
+				v-model="filterValue"
+				type="checkbox"
+				class="filter-checkbox"
+				@change="updateFilter('checkbox', filterValue)" />
+			<span>{{ column.label }}</span>
+		</label>
+
+		<select
+			v-else-if="getFilterType(column) === 'select'"
+			v-model="filterValue"
+			class="filter-select"
+			@change="updateFilter('select', filterValue)">
+			<option value="">All</option>
+			<option v-for="option in getSelectOptions(column)" :key="option.value || option" :value="option.value || option">
+				{{ option.label || option }}
+			</option>
+		</select>
+
+		<input
+			v-else-if="getFilterType(column) === 'date'"
+			v-model="filterValue"
+			type="date"
+			class="filter-input"
+			@input="updateFilter('date', filterValue)" />
+
+		<div v-else-if="getFilterType(column) === 'dateRange'" class="date-range-filter">
+			<input
+				v-model="filterStartValue"
+				type="date"
+				class="filter-input"
+				@input="updateDateRangeFilter('start', filterStartValue)" />
+			<span class="clear-btn"> - </span>
+			<input
+				v-model="filterEndValue"
+				type="date"
+				class="filter-input"
+				@input="updateDateRangeFilter('end', filterEndValue)" />
+		</div>
+
+		<component
+			v-else-if="getFilterType(column) === 'component' && column.filterComponent"
+			:is="column.filterComponent"
+			:value="filterValue"
+			:column="column"
+			:colIndex="colIndex"
+			:store="store"
+			@update:value="updateFilter('component', $event)" />
+
+		<button v-if="hasActiveFilter" @click="clearFilter" class="clear-btn" title="Clear">×</button>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { ref, computed } from 'vue'
+import { createTableStore } from '../stores/table'
+import type { TableColumn } from '../types'
+
+const { column, colIndex, store } = defineProps<{
+	column: TableColumn
+	colIndex: number
+	store: ReturnType<typeof createTableStore>
+}>()
+
+const filterValue = ref<any>('')
+const filterStartValue = ref<any>('')
+const filterEndValue = ref<any>('')
```

Comment:
Since these are related, I wonder if it's better to use a `reactive` instead?


```suggestion
const dateFilter = reactive({
	startValue: '' as string,
	endValue: '' as string,
})
```

Reply:
Applied!